plenum/server/node.py

```diff
@@ -2338,10 +2326,68 @@ def allLedgersCaughtUp(self):
             logger.info('{}{} does not need any more catchups'
                         .format(CATCH_UP_PREFIX, self),
                         extra={'cli': True})
+
+            self.selection_primaries_after_catchup()
+
+            # set view_change_in_progress to False
             self.no_more_catchups_needed()
-            # select primaries after pool ledger caughtup
-            if not self.view_change_in_progress:
-                self.select_primaries()
+
```

Comment:
no_more_catchups_needed must be after selection, because view_change_in_progress is the only field indicating view_change now